modules/packages/LinearAlgebra.chpl

```diff
@@ -1367,6 +1367,88 @@ proc cholesky(A: [] ?t, lower = true)
 }
 
 
+/* Find the eigenvalues of a real-symmetric/complex-hermitian matrix
+   ``A``. ``A`` must be square.
+
+   The algorithms uses either the lower-triangular (if ``lower`` is ``true``,
+   or upper-triangular part of the matrix only. If ``overwrite`` is
+   true, on exiting, this part
+   of the matrix, including the diagonal is overwritten.
+
+   .. note::
+
+     This procedure currently just returns all eigenvalues.
+     To selectively return certain eigenvalues, the user should call the
+     LAPACK routine directly.
+
+   .. note::
+
+      This procedure depends on the :mod:`LAPACK` module, and will generate a
+      compiler error if ``lapackImpl`` is ``none``.
+
+*/
+proc eigvalsh(A: [] ?t, lower=true, param overwrite=false) throws where (A.domain.rank == 2) && (usingLAPACK) {
+  return eigh(A, lower=lower, overwrite=overwrite, eigvalsOnly=true);
+}
+
+/* Find the eigenvalues and eigenvectors of a real-symmetric/complex-hermitian matrix
+   ``A``. ``A`` must be square.
+
+   The algorithms uses either the lower-triangular (if ``lower`` is ``true``,
+   or upper-triangular part of the matrix only.
+
+   If ``overwrite`` is true, the matrix is overwritten with the eigenvectors and
+   only the eigenvalues are returned, otherwise the original matrix is preserved.
+
+   The eigenvectors are stored in the columns of the returned matrix i.e. ``A[..,i]`` is the
+   ``i``'th eigenvector.
+
+   .. note::
+
+     This procedure currently returns all eigenvalues and eigenvectors.
+     To selectively return certain eigenvalues/eigenvectors, the user should call the
+     LAPACK routine directly.
+
+   .. note::
+
+      This procedure depends on the :mod:`LAPACK` module, and will generate a
+      compiler error if ``lapackImpl`` is ``none``.
+
+*/
+proc eigh(A: [] ?t, lower=true, param eigvalsOnly=false, param overwrite=false) throws where (A.domain.rank == 2) && (usingLAPACK) {
+  const (n,m) = A.shape;
+  if n != m then throw new LinearAlgebraError("Non-square matrix passed to eigh");
+  param nbits = numBits(t);
+  var w = Vector(n, eltType = if isComplexType(t) then real(nbits/2) else t);
+
+  var Acopy = if overwrite then none else A;
+  ref Aref = if overwrite then A else Acopy;
+  var jobz = if !eigvalsOnly then "V" else "N";
+
+  var info : int;
+  const uploStr = if lower then "L" else "U";
+
+  if isComplexType(t) {
+    compilerAssert((nbits==64)||(nbits==128),"LAPACK only supports 64 and 128 bit complex types");
+    info = LAPACK.heev(lapack_memory_order.row_major, jobz, uploStr, Aref, w);
+  } else if isRealType(t) {
+    compilerAssert((nbits==32)||(nbits==64),"LAPACK only supports 32 and 64 bit complex types");
```

Comment:
This error message seems like it should refer to `real` types.

Reply:
Good catch @cassella